File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int l=0,r=heights.length-1;
        int res = 0,area=0;
        while(l<r)
        {
            area = (r-l) * Math.min(heights[r],heights[l]);
            res = Math.max(area,res);
            if(heights[l]<heights[r])
            {
                l++;
            }
            else
            {
                r--;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-1.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        s={}
        n=len(cost)
        def dp(i,s):
            if i >=n:
                return 0
            if i in s:
                return s[i]
            s[i]=cost[i]+min(dp(i+1,s),dp(i+2,s))
            return s[i]
        return min(dp(0,s),dp(1,s))
    